ls1intum/Artemis
# [Exam] Enable ordering of exercise groups

### Checklist

### Motivation and Context
The order of the exercise groups is relevant for the order of exercises of each student exam.

### Description
- [x] The exercise groups will be now retrieved via the exam so that the order will be available in the client.
- [x] I've created a new server route to save the new exercise group order. The route checks whether only the order of existing exercise groups was changed.
- [x] I've added a Spring integration test for the new route.
- [x] I've ensured that on deletion of an exercise group it will be deleted via the exam so that the `exercise_group_order` will stay continuous (this fixes an error that prevents an exam from being deleted).

### Steps for Testing
1. Create an exam
2. Add multiple exercise groups
3. Change the order
4. Refresh the page and check that the defined order is saved

If you test locally, you can also check via `SELECT * FROM exercise_group` that the values in the column `exercise_group_order` for the exam are continuous even after you have deleted an exercise group.

### Screenshots
![order-exercise-groups](https://user-images.githubusercontent.com/15168372/85459580-9d809300-b5a2-11ea-8c53-21997f0cc117.gif)
